### ISSUE TITLE
explicit-module-boundary-types with declared type and default value 

### ISSUE THREAD

**Issue by Conaclos** [denoland/deno_lint#1048]
Hi!

This could be related to https://github.com/denoland/deno_lint/issues/1018. However this is a bit different because the type annotation is present. deno-lint seems to ignore it.

### Lint Name

[explicit-module-boundary-types](https://lint.deno.land/?q=&all=on#explicit-module-boundary-types)

### Code Snippet

```json
{
  "lint": {
    "files": {
      "include": ["src/"]
    },
    "rules": {
      "tags": ["recommended"],
      "include": ["explicit-module-boundary-types"]
    }
  }
}
```

```ts
export interface Conf {
    readonly value: number
}

export function init(conf: Partial<Conf> = {}): Conf {
    return { value: 1, ...conf }
}
```

### Expected Result

Should pass the lint rule. 

### Actual Result

```
error[explicit-module-boundary-types]: All arguments should be typed
   |
 5 | export function init(conf: Partial<Conf> = {}): Conf {
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Add types to all the function arguments
```

### Version

deno 1.21.3 (release, x86_64-unknown-linux-gnu)
v8 10.0.139.17
typescript 4.6.2

I also tested with @node-rs/denolint 1.13.1 (release, x86_64-unknown-linux-gnu)
(The previous diagnostic comes from there)

**Comment by Conaclos** [comment]
The issue was resolved in https://github.com/denoland/deno_lint/issues/1048
